arelle/Updater.py

```diff
@@ -6,28 +6,51 @@
 """
 from __future__ import annotations
 
-import gettext
+import datetime
 import os
 import subprocess
 import sys
 import threading
 import tkinter.messagebox
-import typing
-from datetime import date
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
 
 import regex
 
 from arelle import Version
+from arelle.typing import TypeGetText
 
-if typing.TYPE_CHECKING:
+if TYPE_CHECKING:
     from arelle.CntlrWinMain import CntlrWinMain
 
-_ = gettext.gettext
+_: TypeGetText
 
 _MESSAGE_HEADER = "arelle\u2122 - Updater"
 _ISO_DATE_PATTERN = regex.compile(
     r"(?P<date>(?P<year>[0-9]{4})-(?P<month>0[1-9]|1[0-2])-(?P<day>0[1-9]|[12][0-9]|3[01]))"
 )
+_SEMVER_PATTERN = regex.compile(
+    r"(?P<semver>(?P<major>[0-9]+)\.(?P<minor>[0-9]+)\.(?P<patch>[0-9]+))"
```

Comment:
I'm not super sure whats going on here but I assume this is always going to reach out to a website so it won't need to match `9.2.2.dev1128+g85b8787b.d20220928` for example, correct?